cmd/adminkubeconfig/kubetoken.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+func parseTokenResponse(location string) (string, error) {
+	locURL, err := url.Parse(location)
+	if err != nil {
+		return "", err
+	}
+
+	for _, param := range strings.Split(locURL.Fragment, "&") {
+		nameValue := strings.Split(param, "=")
+		if nameValue[0] == "access_token" {
+			return nameValue[1], nil
+		}
+	}
+	return "", fmt.Errorf("token not found in response")
+}
+
+func getTokenURLFromConsoleURL(consoleURL string) (*url.URL, error) {
+	tokenURL, err := url.Parse(strings.Replace(consoleURL, "console-openshift-console.apps.", "oauth-openshift.apps.", 1))
```

Comment:
if you're using url.Parse (good), then just set the host field or what have you on tokenURL below instead of strings.Replace?